### PR TITLE
Restructure config targets

### PR DESCRIPTION
**Pull request by NeroBurner**
I modernized the cmake configure files by adding namespaces and include interfaces

- use namespace SuiteSparse:: for cmake targets (for example SuiteSparse::suitesparseconfig)
- add include directory to install targets
- install to GNUInstallDirs
    - install libs to `lib`
- install cmake config files into lib/cmake/suitesparse-<VERSION>
- don't install UseSuiteSparse.cmake
- move config templates into `cmake` folder
- config file provides `SuiteSparse_INCLUDE_DIRS` and `SuiteSparse_LIBRARIES`

other fixes:
- remove trailing spaces in top CMakeLists file
- fix GKlib try_compile filepath

edit:
an install package now has the following structure:
```
- include/suitesparse
- lib
    - <libraries>
    - cmake/suitesparse-<VERSION>
        - suitesparse-config.cmake
        - suitesparse-config-version.cmake
        - suitesparse-targets.cmake
        - suitesparse-targets-noconfig.cmake
```

edit 2:
lower case suitesparse-config files

### CODE REVIEW

File: cmake/SuiteSparse-config-install.cmake.in [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Compute locations from <prefix>/@{LIBRARY_DIR@/cmake/suitesparse-<v>/<self>.cmake
+get_filename_component(_SuiteSparse_SELF_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)
+get_filename_component(_SuiteSparse_PREFIX "${_SuiteSparse_SELF_DIR}" PATH)
+get_filename_component(_SuiteSparse_PREFIX "${_SuiteSparse_PREFIX}" PATH)
+get_filename_component(_SuiteSparse_PREFIX "${_SuiteSparse_PREFIX}" PATH)
+
+# Load the LAPACK package with which we were built.
+find_package(LAPACK CONFIG)
```
Comment (ruslo) [inline cmake/SuiteSparse-config-install.cmake.in]: I saw `add_library(lapack SHARED IMPORTED)` in code. Are you sure this is working? I expect `LAPACKConfig.cmake` somewhere.

File: cmake/SuiteSparse-config-install.cmake.in [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+# Load targets from the install tree.
+if(NOT TARGET SuiteSparse::suitesparseconfig)
+  include(${_SuiteSparse_SELF_DIR}/SuiteSparse-targets.cmake)
```
Comment (ruslo) [inline cmake/SuiteSparse-config-install.cmake.in]: Can be just `include("${CMAKE_CURRENT_LIST_DIR}/SuiteSparse-targets.cmake")`, no need for `if(NOT TARGET ...)`.

File: cmake/SuiteSparse-config-install.cmake.in [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+endif()
+
+# Report SuiteSparse header search locations.
+set(SuiteSparse_INCLUDE_DIRS ${_SuiteSparse_PREFIX}/include)
```
Comment (ruslo) [inline cmake/SuiteSparse-config-install.cmake.in]: There is no need to add include path explicitly, it will be added automatically because of `target_include_directories(... PUBLIC ...)`.

File: cmake/SuiteSparse-config-install.cmake.in (hunk 1 above)
Comment (NeroBurner) [inline cmake/SuiteSparse-config-install.cmake.in]: yes, those code blocks are used only if no `lapack-config.cmake` is found AND when on Windows.
The `else()` path of `if(LAPACK_FOUND)` is not indented. Maybe this should be changed in another PR

File: cmake/SuiteSparse-config-install.cmake.in (hunk 3 above)
Comment (NeroBurner) [inline cmake/SuiteSparse-config-install.cmake.in]: This line is in for backwards compatibility. I think it does not hurt to leave it in

File: cmake/SuiteSparse-config-install.cmake.in (hunk 2 above)
Comment (NeroBurner) [inline cmake/SuiteSparse-config-install.cmake.in]: fixed

**Comment by NeroBurner** [comment]
rebased on current master (with fPIC patch)

shall I split this PR into smaller chunks?

**Comment by jlblancoc** [comment]
Leave it as is for now, thanks. Will take a look at this one soon and give
you feedback, if needed...

File: CMakeLists.txt [hunk 4]
```diff
@@ -57,20 +60,20 @@ SET(METIS_DIR ${${PROJECT_NAME}_SOURCE_DIR}/metis CACHE PATH "Source directory o
 if(BUILD_METIS)
 	## prepare the installation :
 	## using metis target here is not possible because this target is added in another branch of the CMake structure
-	## TRICK: need to dynamically modify the metis CMakeLists.txt file before it going to parsed... 
+	## TRICK: need to dynamically modify the metis CMakeLists.txt file before it going to parsed...
 	## (very ugly/poor for a metis project get from SCM (git/svn/cvs) but it's works ;) and it doesn't matter if metis was get from .zip)
```
Comment (jlblancoc) [inline CMakeLists.txt]: 👍
